java-checks/src/main/java/org/sonar/java/checks/SwitchLastCaseIsDefaultCheck.java

```diff
@@ -57,7 +58,14 @@ private static boolean getDefaultLabel(SwitchStatementTree switchStatementTree)
   }
 
   private static boolean isDefault(CaseLabelTree caseLabelTree) {
-    return JavaKeyword.DEFAULT.getValue().equals(caseLabelTree.caseOrDefaultKeyword().text());
+    if (equalsDefaultKeyword(caseLabelTree.caseOrDefaultKeyword().text())) {
+      return true;
+    }
+    return caseLabelTree.expressions().stream().anyMatch(expr -> equalsDefaultKeyword(expr.firstToken().text()));
+  }
+
+  private static boolean equalsDefaultKeyword(String text) {
+    return JavaKeyword.DEFAULT.getValue().equals(text);
```

Comment:
What about extracting `JavaKeyword.DEFAULT.getValue()` in a dedicated constant, so it is not recalculated all the time?